Sources/Observer.swift

```diff
@@ -68,6 +84,15 @@ extension Signal {
 			}
 		}
 
+		deinit {
+			if interruptsOnDeinit {
+				// Since `Signal` would ensure that only one terminal event would ever be
+				// sent for any given `Signal`, we do not need to assert any condition
+				// here.
+				action(.interrupted)
```

Comment:
It would mess up the promise of no event being sent after any terminal event, unless we track termimation per observer.

Detaching an observer could use an `interrupted` event though.